src/spec_to_agents/agents/venue_specialist.py

```diff
@@ -1,24 +1,35 @@
 # Copyright (c) Microsoft. All rights reserved.
 
-from agent_framework import ChatAgent, MCPStdioTool
-from agent_framework.azure import AzureAIAgentClient
+from agent_framework import BaseChatClient, ChatAgent, ToolProtocol
+from dependency_injector.wiring import Provide, inject
 
-from spec_to_agents.models.messages import SpecialistOutput
 from spec_to_agents.prompts import venue_specialist
 from spec_to_agents.tools import web_search
 
 
-def create_agent(client: AzureAIAgentClient, mcp_tool: MCPStdioTool | None) -> ChatAgent:
+@inject
+def create_agent(
+    client: BaseChatClient = Provide["client"],
```

Comment:
I've never seen python DI done this way, but learned something new - looks good! Will be sure to document in the docs as it'll likely confuse students.